src/form/Form.tsx

```diff
@@ -484,22 +498,40 @@ function Form({
     }, 750),
     [activeStep?.id, formRef, validateElements]
   );
+
+  // Debouncing the hide if referenced field changes check call to rate limit re-renders
+  const debouncedHideRuleCheck = useCallback(
+    debounce((changedFields: [string, unknown][]) => {
+      // Re-render field/element if any field has changed that is
+      //  referenced in the field/element's hide if rule.
+      if (changedFields.some(([key]) => hideIfFieldReferences.has(key)))
+        setRender((render) => !render);
+    }, 500),
+    [hideIfFieldReferences, setRender, render]
+  );
+
   useEffect(() => {
     return () => {
       debouncedValidate.cancel();
+      debouncedHideRuleCheck.cancel();
     };
-  }, [debouncedValidate]);
+  }, [debouncedValidate, debouncedHideRuleCheck]);
 
   const updateFieldValues = (newFieldValues: any, rerender = true) => {
     const entries = Object.entries(newFieldValues);
-    const noChange = entries.every(([key, val]) => fieldValues[key] === val);
-    if (noChange) return false;
+    const changedEntries = entries.filter(
+      ([key, val]) => fieldValues[key] !== val
+    );
+    if (!changedEntries.length) return false;
 
     const empty = entries.some(([key, val]) => !val || !fieldValues[key]);
     Object.assign(fieldValues, newFieldValues);
+
     // Always rerender from empty state for display purposes
     if (rerender || empty) setRender((render) => !render);
 
+    debouncedHideRuleCheck(changedEntries);
```

Comment:
how does this affect performance? especially while typing in text fields on large steps

Reply:
also wouldn't this debouncing potentially prevent renreders that need to happen?